### DISCUSSION TITLE
Sqlalchemy 1.3 and 1.4+ differences

### DISCUSSION THREAD

**Discussion by reuvenstr** [sqlalchemy/sqlalchemy#11921] [Usage Questions] [answered]
Hi, 
I'm trying to upgrade my code base to use SQLalchemy 1.4, and one thing that started to break is` json.dumps` on `session.query(model.col).all()` results, it's of type `Row`, and not `result` as it was previously?

Does someone know how I can achieve the same behavior? It only occurs if part of the model is requested, and not all columns.

Thanks ahead

**Comment by CaselIT** [comment, marked as answer]
Hi,

The changes are detailed here:
- https://docs.sqlalchemy.org/en/14/changelog/migration_14.html
- https://docs.sqlalchemy.org/en/14/changelog/changelog_14.html

In particular https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#change-4710-core

Your comment if very light on details, but I suppose you want to serialize things as a list dict, do do that you could do `json.dums([row._asdict() for row in session.query(model.col).all()])`

**Reply by reuvenstr** [reply]
Looks like this one explains my struggling 
https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#change-4710-orm

That's how I fixed it, just hoped that would be a more robust solution instead of `dict(row)` on every row.

Thanks a lot!
